src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java

```diff
@@ -112,69 +116,47 @@ private void sendRequest(String stringEvent, boolean isAdminEvent) {
-                sb.append(e.getValue());
-                sb.append("\", ");
+        try {
+            HashMap<String, Object> resultMap = new HashMap<>();
+            resultMap.put("type", event.getType());
+            resultMap.put("realmId", event.getRealmId());
+            resultMap.put("clientId", event.getClientId());
+            resultMap.put("userId", event.getUserId());
+            resultMap.put("ipAddress", event.getIpAddress());
+
+            String eventError = event.getError();
+            if (eventError != null && eventError.length() > 0) {
+                resultMap.put("error", eventError);
+            }
+            Map<String, String> details = event.getDetails();
+            if (details != null && !details.isEmpty()) {
+                resultMap.put("details", details);
             }
-        sb.append("}}");
+            return objectMapper.writeValueAsString(resultMap);
+        } catch (JsonProcessingException e) {
+            return "";
         }
-
-        return sb.toString();
     }
 
-
     private String toString(AdminEvent adminEvent) {
-        StringBuilder sb = new StringBuilder();
-
-        sb.append("{\"type\": \"");
-        sb.append(adminEvent.getOperationType());
-        sb.append("\", \"realmId\": \"");
-        sb.append(adminEvent.getAuthDetails().getRealmId());
-        sb.append("\", \"clientId\": \"");
-        sb.append(adminEvent.getAuthDetails().getClientId());
-        sb.append("\", \"userId\": \"");
-        sb.append(adminEvent.getAuthDetails().getUserId());
-        sb.append("\", \"ipAddress\": \"");
-        sb.append(adminEvent.getAuthDetails().getIpAddress());
-        sb.append("\", \"resourcePath\": \"");
-        sb.append(adminEvent.getResourcePath());
-        sb.append("\", \"resourceType\": \"");
-        sb.append(adminEvent.getResourceType());
-        sb.append("\"");
-
-        if (adminEvent.getError() != null) {
-            sb.append(", \"error\": \"");
-            sb.append(adminEvent.getError());
-            sb.append("\"");
+        try {
+            HashMap<String, Object> resultMap = new HashMap<>();
+            resultMap.put("type", adminEvent.getOperationType());
+            resultMap.put("realmId", adminEvent.getAuthDetails().getRealmId());
+            resultMap.put("clientId", adminEvent.getAuthDetails().getClientId());
+            resultMap.put("userId", adminEvent.getAuthDetails().getUserId());
+            resultMap.put("ipAddress", adminEvent.getAuthDetails().getIpAddress());
+            resultMap.put("resourcePath", adminEvent.getResourcePath());
+            resultMap.put("resourceType", adminEvent.getResourceType());
+            String eventError = adminEvent.getError();
+            if (eventError != null && eventError.length() > 0) {
+                resultMap.put("error", eventError);
+            }
+            return objectMapper.writeValueAsString(resultMap);
+        } catch (JsonProcessingException e) {
+            return "";
```

Comment:
how often does this happen? can you log a warning here, to avoid a silent error?